hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/setapproval/SetApprovalForAllTranslator.java

```diff
@@ -54,8 +53,9 @@ public SetApprovalForAllTranslator(final SetApprovalForAllDecoder decoder) {
      */
     @Override
     public boolean matches(@NonNull final HtsCallAttempt attempt) {
-        return selectorMatches(attempt, SET_APPROVAL_FOR_ALL)
-                || (attempt.isTokenRedirect() && selectorMatches(attempt, ERC721_SET_APPROVAL_FOR_ALL));
+        return attempt.isTokenRedirect()
```

Comment:
This expression is _not_ the same as what it replaces.  Seems like it is in fact correct now, but please re-check.

Reply:
I did check it with the same concerns but it seems to be correct.